azure/scope/cluster.go

```diff
@@ -297,6 +297,36 @@ func (s *ClusterScope) GroupSpec() azure.ResourceSpecGetter {
 	}
 }
 
+// SubnetSpecs returns the subnets specs.
+func (s *ClusterScope) VnetPeeringSpecs() []azure.VnetPeeringSpec {
+	peeringSpecs := []azure.VnetPeeringSpec{}
```

Comment:
Consider preallocating the slice rather than appending.

```suggestion
	peeringSpecs := make([]azure.VnetPeeringSpec, 2 * len(s.Vnet().Peerings))
```